mono/metadata/w32handle.c

```diff
@@ -199,7 +236,7 @@ mono_w32handle_unlock_signal_mutex (void)
 }
 
 void
-mono_w32handle_lock_handle (gpointer handle)
+mono_w32handle_lock_handle (gpointer handle, MonoW32HandleType expected_types)
```

Comment:
Why add it here if it's not used? You could use the expected_types as part of `mono_w32handle_ref` too?